sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -163,6 +163,24 @@ case class DataSourceResolution(
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
 
+    case ShowTablesStatement(None, pattern) =>
+      defaultCatalog match {
+        case Some(_) =>
+          throw new AnalysisException(
+            "v2 catalog doesn't support getCurrentDatabase yet, " +
+              "so the default database name cannot be deduced.")
+        case None =>
+          ShowTablesCommand(None, pattern)
+      }
+
+    case plan @ ShowTablesStatement(Some(namespace), pattern) =>
+      val CatalogObjectIdentifier(maybeCatalog, identifier) = namespace
```

Comment:
I think we may need a different matcher. I noted below that this passes an identifier instead of `Array[String]`, but I also realized that the identifier here must have at least a table name. So `SHOW TABLES FROM catalog` would produce `CatalogObjectIdentifer(None, Identifer("catalog"))`. I don't think that's correct. We should probably add a `CatalogNamespace` extractor to handle this. Then you'd get an `Array[String]` from it and could base the plans on passing a real namespace.

Reply:
Ah OK. I didn't know `SHOW TABLES FROM catalog` should resolve to "`SHOW TABLES` using `catalog`." What if `catalog` is a database name? In that case, what should be the default behavior.

Reply:
I pushed out a new change assume what you meant was 1). Please let me know if I misunderstood.

Reply:
Catalog names always take precedence over namespaces, so the collision should be interpreted as a catalog name. To avoid this, users can explicitly add the catalog.

Reply:
Got it, updated the code/tests reflecting this.